packages/webdriverio/src/utils/findStrategy.ts

```diff
@@ -168,15 +168,22 @@ export const findStrategy = function (selector, isW3C, isMobile) {
     }
     case 'name': {
         if (isMobile || !isW3C) {
+            const match = selector.match(/^\[name=("|')([a-zA-z0-9\-_.@=[\] ']+)("|')]$/)
+            if (!match) {
+                throw new Error('InvalidSelectorStrategy') // ToDo: move error to wdio-error package
+            }
             using = 'name'
-            value = selector.match(/^\[name=("|')([a-zA-z0-9\-_.@=[\] ']+)("|')]$/)[2]
+            value = match[2]
         }
         break
     }
     case 'xpath extended': {
         using = 'xpath'
         const match = selector.match(new RegExp(XPATH_SELECTOR_REGEXP.map(rx => rx.source).join('')))
-        const PREFIX_NAME = { '.': 'class', '#': 'id' }
+        if (!match) {
+            throw new Error('InvalidSelectorStrategy') // ToDo: move error to wdio-error package
```

Comment:
same

Reply:
Taken care in another thread: https://github.com/webdriverio/webdriverio/pull/6023#discussion_r510838614